src/GitHub.App/Services/ModelService.cs

```diff
@@ -157,22 +157,33 @@ public IObservable<AccountCacheItem> GetUserFromCache()
             var keyobs = GetUserFromCache()
                 .Select(user => string.Format(CultureInfo.InvariantCulture, "{0}|{1}:{2}", CacheIndex.PRPrefix, user.Login, repo.Name));
 
-            var source = Observable.Defer(() => keyobs
-                .SelectMany(key =>
-                    hostCache.GetAndFetchLatestFromIndex(key, () =>
+            var source = Observable.Defer(() =>
+            {
+                var collectionIsDisposed = false;
+
+                return keyobs
+                    .SelectMany(key => hostCache.GetAndFetchLatestFromIndex(key, () =>
                         apiClient.GetPullRequestsForRepository(repo.CloneUrl.Owner, repo.CloneUrl.RepositoryName)
-                                 .Select(PullRequestCacheItem.Create),
-                        item =>
-                        {
-                            // this could blow up due to the collection being disposed somewhere else
-                            try { collection.RemoveItem(Create(item)); }
-                            catch (ObjectDisposedException) { }
-                        },
-                        TimeSpan.Zero,
-                        TimeSpan.FromDays(7))
-                )
-                .Select(Create)
-            );
+                            .Select(PullRequestCacheItem.Create), item =>
```

Comment:
Could you move item to the next line and unindent it, as well as the lambda block for it? This indentation makes it look like the lambda block is scoped at the same level as the `.Select` statement, which is misleading.